src/provider/utility/provider.utility.ts

```diff
@@ -2,15 +2,41 @@ import { stringToBase64 } from '@gnolang/tm2-js-client';
 
 /**
  * Prepares the VM ABCI query params by concatenating them
- * with a newline separation and encoding them to base64
+ * with new line or "." or ":" characters separation and encoding them to base64
+ * `evaluateExpression` uses the "." character to separate parameters.
+ * `getRenderOutput` uses the ":" character to separate parameters.
  * @param {string[]} params the params for the ABCI call
+ * @param {string} separator the separator for ABCI call parameters (default: "\n")
  */
-export const prepareVMABCIQuery = (params: string[]): string => {
+export const prepareVMABCIQuery = (
+  params: string[],
+  separator = '\n'
```

Comment:
I think we should always require a separator, since we don't use `\n` anywhere anymore

Reply:
We are already using the functions of `prepareVMABCIQuery` without separator.
Example: https://github.com/gnolang/gno-js-client/blob/main/src/provider/jsonrpc/jsonrpc.ts#L46-L56

I would like to declare an additional function prepareVMABCIQueryWithSeparator, could you take a look at it?
https://github.com/gnolang/gno-js-client/pull/129/commits/76dc6d174fc8b836d5b112da472d00ab3e7d0588

Reply:
Ah yes, I forgot about those 2 that only have single params